frontend/src/views/GameView.vue

```diff
@@ -1,51 +1,156 @@
 <template>
-	<div class="page_box_wrapper">
-		<div class="page_box">
-			<h1>This is a game page</h1>
+	<div >
+
+		<div class="page_box_wrapper">
+			<PlayerNames player_left="renoster" player_right="aardwolf" />
+
+				<div class="page_box">
+					<div class="gameHeader space-between">
+						<p class="playerOneSore"> 5 </p>
+						<p class="pongHeader">PONG</p>
+						<p class="playerTwoScore"> 3 </p>
+					</div>
+
+					<canvas
+					ref="GameRef"
+					id="GameCanvas"
+					class="gameBlock"
+					width="1200"  
+					height="960">
+					</canvas>
+
+				</div>
 		</div>
-		<p class="table table-striped table-sm">
-			All Games:
-			<tbody>
-				<tr v-for="game in gameStore.allGames" :key="game.id">
-					<td> Game id: {{ game.id }} </td>
-					<td> Score player one: {{ game.score_player_one }} </td>
-					<td> Score player two: {{ game.score_player_two }} </td>
-					<td>Is active: {{ game.is_active }}</td>
-				</tr>
-				
-			</tbody>
-
-		</p>
-		<p class="table table-striped table-sm">
-			All My Games:
-			<tbody>
-				<tr v-for="game in gameStore.getMyGames" :key="game.id">
-					<td>Game id: {{ game.id }}</td>
-					<td>Score player one: {{ game.score_player_one }}</td>
-					<td>Score player two: {{ game.score_player_two }}</td>
-					<td>Is active: {{ game.is_active }}</td>
-				</tr>
-				
-			</tbody>
-			
-		</p>
 	</div>
 </template>
 
-<!--  THIS SCRIPT IS ONLY TO SHOW A WORKING GAME STORE SETUP-->
 <script lang="ts">
 import { useGameStore } from '@/stores/gameStore';
 import { defineComponent } from 'vue';
+import PlayerNames from '@/components/game-info/PlayerNames.vue'
+
+interface DataObject {
+	context: CanvasRenderingContext2D
+}
 
 export default defineComponent({
 	name: "GameView",
+	components: {
+		PlayerNames
+	},
+
+	data(): DataObject {
+		return {
+			context: {} as CanvasRenderingContext2D
+		}
+	},
 
 	setup(){
 		const gameStore = useGameStore();
 		gameStore.refreshData();
 		return {
 			gameStore,
 		}
-	}
+	},
+	computed: {
+		// typescript wants values used from html to have a type
+		width() {
+			return (this.$refs.GameRef as HTMLCanvasElement).width;
+		},
+		height() {
+			return (this.$refs.GameRef as HTMLCanvasElement).height;
+		},
+	},
+	methods: {
+		// https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillRect
+		drawMiddleLine() {
+			const widthStart: number = this.width / 2;
+			const heightStart: number = 0;
+			const lineWidth: number = 4;
+			const lineHeight: number = this.height; 
+			
+			this.context.fillRect(
+					widthStart,
+					heightStart,
+					lineWidth,
+					lineHeight
+				);
+			},
+			
+		//  https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/arc
+		drawBall() {
+			const startingX: number = this.width / 2 - 100;
+			const startingY: number = this.height / 2 - 100;
+			const radius: number = 15;
```

Comment:
If we want these fully relative to canvas, the `- 100` part should also some percentage based value (`this.width/100*10` for 10% or something like that)

Reply:
Indeed, adde methods width/heightPercentage(perc: number) that returns the wanted percentage value

Reply:
perfect!